internal/cli/users.go

```diff
@@ -22,8 +71,347 @@ func usersCmd(cli *cli) *cobra.Command {
+
+			if len(inputs.sort) == 0 {
+				queryParams = append(queryParams, management.Query(auth0.StringValue(&inputs.query)))
+			} else {
+				queryParams = append(queryParams,
+					management.Query(auth0.StringValue(&inputs.query)),
+					management.Parameter("sort", auth0.StringValue(&inputs.sort)),
+				)
+			}
+
+			if err := ansi.Waiting(func() error {
+				var err error
+				search, err = cli.api.User.Search(queryParams...)
+				return err
+			}); err != nil {
+				return fmt.Errorf("An unexpected error occurred: %w", err)
+			}
+
+			cli.renderer.UserSearch(search.Users)
+			return nil
+		},
+	}
+	userQuery.RegisterString(cmd, &inputs.query, "")
+	userSort.RegisterString(cmd, &inputs.sort, "")
+
+	return cmd
+}
+
+func createUserCmd(cli *cli) *cobra.Command {
+	var inputs struct {
+		Connection string
+		Email      string
+		Password   string
+		Username   string
+		Name       string
+	}
+
+	cmd := &cobra.Command{
+		Use:   "create",
+		Args:  cobra.NoArgs,
+		Short: "Create a new user",
+		Long:  "Create a new user",
+		Example: `auth0 users create 
+auth0 users create --name "John Doe" 
+auth0 users create -n "John Doe" --email john@example.com
+auth0 users create -n "John Doe" --e john@example.com --connection "Username-Password-Authentication"`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Select from the available connection types
+			// Users API currently support  database connections
+			if err := userConnection.Select(cmd, &inputs.Connection, cli.connectionPickerOptions(), nil); err != nil {
+				return err
+			}
+
+			// Prompt for user's name
+			if err := userName.Ask(cmd, &inputs.Name, nil); err != nil {
+				return err
+			}
+
+			// Prompt for user email
+			if err := userEmail.Ask(cmd, &inputs.Email, nil); err != nil {
+				return err
+			}
+
+			////Prompt for user password
+			inputs.Password = prompt.Password(inputs.Password)
+
+			// The getConnReqUsername returns the value for the requires_username field for the selected connection
+			// The result will be used to determine whether to prompt for username
+			conn := cli.getConnReqUsername(auth0.StringValue(&inputs.Connection))
+			requireUsername := auth0.BoolValue(conn)
+
+			var a *management.User
+
+			// Prompt for username if the requireUsername is set to true
+			// Load values including the username's field into a fresh users instance
+			// Else block loads values without username for connections with requireUsername set to false
+			if requireUsername {
+				if err := userUsername.Ask(cmd, &inputs.Username, nil); err != nil {
+					return err
+				}
+				a = &management.User{
+					Connection: &inputs.Connection,
+					Email:      &inputs.Email,
+					Name:       &inputs.Name,
+					Username:   &inputs.Username,
+					Password:   &inputs.Password,
+				}
+			} else {
+				a = &management.User{
+					Connection: &inputs.Connection,
+					Email:      &inputs.Email,
+					Name:       &inputs.Name,
+					Password:   &inputs.Password,
+				}
```

Comment:
This could be simplified by first creating the instance without the username, and then adding it later if necessary.